.github/scripts/promote_branch.sh

```diff
@@ -0,0 +1,214 @@
+#     script will not create a PR to add more content to the staging branch. This can be overridden with
+#     --force-to-staging
+#   - If promoting to production and the content has not been in the staging branch for at least 7 days,
+#     the script will exit without creating a PR. Content is expected to sit in staging for at least a week
+#     to provide sufficient testing time. This can be overridden with --override
+#
+# Prerequisities:
+#   - An environment variable GITHUB_TOKEN is defined that provides access to the users repository. See
+#     https://github.com/redhat-appstudio/release-service-utils/blob/main/ci/promote-overlay/README.md#setup for help.
+#   - git, patch and yq are installed.
+
+set -e
+
+# GitHub repository details
+ORG="redhat-appstudio"
+REPO="release-service-catalog"
+
+OPTIONS=$(getopt --long "target-branch:,fork-owner:,force-to-staging,override,debug,help" -o "tgt:,fo:,h" -- "$@")
+eval set -- "$OPTIONS"
+while true; do
+    case "$1" in
+        -tgt|--target-branch)
+            TARGET_BRANCH="$2"
+            shift 2
+            ;;
+        -fo|--fork-owner)
+            FORK_OWNER="$2"
+            shift 2
+            ;;
+        --force-to-staging)
+            FORCE_TO_STAGING="true"
+            shift
+            ;;
+        --override)
+            OVERRIDE="true"
+            shift
+            ;;
+        --debug)
+            DEBUG="true"
+            shift
+            ;;
+        -h|--help)
+            print_help
+            exit
+            ;;
+        --)
+            shift
+            break
+            ;;
+        *) echo "Error: Unexpected option: $1" % >2
+    esac
+done
+
+print_help(){
+    echo "Usage: $0 --target-branch branch --fork-owner owner [--force-to-staging] [--override] [--debug]"
+    echo
+    echo "  --target-branch:    The name of the branch to be promoted. Options are staging and production."
+    echo "  --fork-owner:       The name of the owner of your release-service-catalog fork in Github."
+    echo "  --force-to-staging: If passed, allow promotion to staging even if staging and production differ."
+    echo "  --override:         If passed, allow promotion to production even if the change has not been in staging for one week."
+    echo "  --debug:            For debugging purposes, do everything except for creating the PR and deleting the temp repo."
+    echo
+    echo "  --target-branch and --fork-owner have to be specified."
+}
+
+update_taskGitRevision_in_pipelines() {
+    TARGET=$1
+    
+    for PIPELINE in $(ls pipelines)
+    do
+        PIPELINE_FILE=pipelines/$PIPELINE/${PIPELINE}.yaml
+        yq . $PIPELINE_FILE > ${PIPELINE_FILE}.1
+        yq '.spec.params[] |= select(.name == "taskGitRevision") |= .default = "'$TARGET'"'\
+            $PIPELINE_FILE > ${PIPELINE_FILE}.2
+        diff ${PIPELINE_FILE}.1 ${PIPELINE_FILE}.2 > ${PIPELINE_FILE}.diff || [ $? -gt 1 ] && exit $?
```

Comment:
Just curious, what's the `$?` for? I guess it's because there likely will be differences, which will return 1 and thus the script would fail here, right?

Reply:
Exactly

Reply:
`diff` returns 0 for no change, 1 if there are changes, and 2 if there are other issues. So I still wanted to fail if `$?` is greater than 1

Reply:
Cool, thanks, makes sense.